### DISCUSSION TITLE
Shipping-Tax calculation vendure?

### DISCUSSION THREAD

**Discussion by trnc-ck** [vendurehq/vendure#2155] [General]
hey,

we're currently evaluating node-based commderce frameworks for b2b shops. How do you calculate shipping taxes at the moment? As we and our clients reside in the EU (e.g. Austria), the shipping tax rate is a combination of the product tax-rates.

we also looked at medusa, which seems to calculate the shipping tax just on the region standard-rate.

and, do you have a preview-screenshot of the new admin interface? we're quite excited, as the ui/ux is very important for our clients.

thanks!

**Comment by michaelbromley** [comment]
Hi!

The calculation of shipping taxes is completely configurable via the [ShippingCalculator class](https://www.vendure.io/docs/typescript-api/shipping/shipping-calculator/). By default, the calculator we ship with uses a fixed tax rate that is set in the admin ui, so you would need to create a custom calculator which inspects the tax rates of the current order lines and returns the appropriate rate for the shipping.

I'm working on the new Admin UI right now, but here are some screenshots from our designs:

![image](https://user-images.githubusercontent.com/6275952/235614869-840ed02d-f87c-43e4-b8ba-f5040555b9f5.png)

![image](https://user-images.githubusercontent.com/6275952/235615216-1d5572b2-2400-4f0d-b2e4-27cab150263e.png)
